### PR TITLE
[master] Porting #50505 to master

### PR DESCRIPTION
**Pull request by garethgreenaway**
Porting #50505 to master

### CODE REVIEW

**Review by DmitryKuzmenko — APPROVED** [review]

**Review by dwoz — CHANGES_REQUESTED** [review]
Looks like some of the test failures are related.

File: salt/minion.py [hunk 1]
```diff
@@ -2219,34 +2219,29 @@ def manage_schedule(self, tag, data):
...
-            self.schedule.save_schedule()
-        elif func == 'get_next_fire_time':
-            self.schedule.get_next_fire_time(name)
+        funcs = {'delete': ('delete_job', (name, persist)),
+                 'add': ('add_job', (schedule, persist)),
+                 'modify': ('modify_job',
+                            (name, schedule, persist)),
+                 'enable': ('enable_schedule', ()),
+                 'disable': ('disable_schedule', ()),
+                 'enable_job': ('enable_job', (name, persist)),
```
Comment (DmitryKuzmenko) [inline salt/minion.py]: Missing function here.
`'run_job': ('run_job', (name,)),`

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]

**Comment by DmitryKuzmenko** [comment]
@garethgreenaway could you please update this PR?

File: salt/minion.py (hunk 1 above)
Comment (DmitryKuzmenko) [reply salt/minion.py]: I've fixed that during resolving merge conflicts.

**Review by dwoz — APPROVED** [review]

**Pull request merged**